napari/_vispy/vispy_text_layer.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+from copy import copy
+from vispy.scene.visuals import Line, Compound
+from vispy.scene.visuals import Text as TextNode
+from .markers import Markers
+from vispy.visuals.transforms import ChainTransform
+
+from ..layers import Text
+from .vispy_base_layer import VispyBaseLayer
+import numpy as np
+
+
+class VispyTextLayer(VispyBaseLayer):
+    _highlight_color = (0, 0.6, 1)
+    _highlight_width = 1.5
+
+    _OUTLINE_MARKERS_NODE_INDEX = 0
+    _TEXT_NODE_INDEX = 1
+    _HIGHLIGHT_TEXT_NODE_INDEX = 2
+
+    def __init__(self, layer):
+        # Create a compound visual with the following thress subvisuals:
+        # Lines: The lines of the interaction box used for highlights.
+        # Text: The text to be diplayed.
+        # Text: Highlighted text to show which text is selected.
+        node = Compound([Line(), TextNode(), TextNode()])
+
+        super().__init__(layer, node)
+
+        self.layer.events.rotation.connect(lambda e: self._on_data_change())
+        self.layer.events.font_size.connect(lambda e: self._on_data_change())
+        self.layer.events.text_color.connect(lambda e: self._on_data_change())
+        self.layer.events.highlight.connect(
+            lambda e: self._on_highlight_change()
+        )
+
+        self.layer.dims.events.ndisplay.connect(
+            lambda e: self._on_display_change()
+        )
+
+        self._on_display_change()
+
+    def _on_display_change(self):
+        parent = self.node.parent
+        self.node.transforms = ChainTransform()
+        self.node.parent = None
+
+        if self.layer.dims.ndisplay == 2:
```

Comment:
Right now entering 3D mode gives me errors for both the `add_text.py` and `nD_text.py`, there should be tests inside `tests/test_viewer.py` and `tests/test_napari.py` and `tests/components/test_viewer_model.py` that include both 2D and nD examples and `viewer.dims.ndim = 2` and `viewer.dims.ndim = 3` as is done for our other layers

For the `nD_text.py` when switching to 3D rendering I think you'll need to turn off some of the get_value / hitbox stuff, which I think is turned off in the shapes layer.

Reply:
I fixed the 3D rendering. Would you mind taking a look and letting me know if you like the behavior? It seems like the text is presented at a fixed orientation to the viewer. This seems like the best way to handle 3D text to me, but I am happy to turn it off if we think it is strange. 

Great point about the lack of 3D tests. I will add those.

Reply:
Tests added!